### PR TITLE
Adding tests for DRILL-4743: HashJoin's not fully parallelized in que…

### PR DESCRIPTION
**Pull request by rhou1**
…ry plan.

Modified Test Automation Framework to extract rowcounts in this testing.

Hi Abhishek,

Can you please review this?

Thanks.

--Robert

### CODE REVIEW

File: framework/resources/Functional/filter/selectivity/negative/q1.res [hunk 1]
```diff
@@ -0,0 +1 @@
+.*VALIDATION ERROR: Option planner.filter.min_selectivity_estimate_factor must be between 0.000000 and 1.000000.
```
Comment (Agirish) [inline framework/resources/Functional/filter/selectivity/negative/q1.res]: Can you refer to expected files in other negative tests - there seems to be an error code on the first line.

File: framework/resources/Functional/filter/selectivity/plan/plan.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    "description": "Filter selectivity tests",
+    "categories": [ "smoke" ],
+    "matrices": [
+        {
+            "query-file": ".*.sql",
+            "schema": "dfs.drillTestDirTpch01Parquet",
+            "output-format": "tsv",
+            "expected-file": ".*.res",
+            "verification-type": [
+                "regex-compare"
```
Comment (Agirish) [inline framework/resources/Functional/filter/selectivity/plan/plan.json]: Can we call this "filter-ratio" or something on those lines. This for now is specific to the filter selectivity usecase.

File: framework/resources/Functional/filter/selectivity/plan/q1.res [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.*Screen.*
```
Comment (Agirish) [inline framework/resources/Functional/filter/selectivity/plan/q1.res]: This looks like a regular explain plan verification right? It should be outside of the filter selectivity tests.

File: framework/resources/Functional/filter/selectivity/negative/q1.res (hunk 1 above)
Comment (rhou1) [inline framework/resources/Functional/filter/selectivity/negative/q1.res]: I have added the error code.

File: framework/resources/Functional/filter/selectivity/plan/plan.json (hunk 2 above)
Comment (rhou1) [inline framework/resources/Functional/filter/selectivity/plan/plan.json]: Done.

File: framework/resources/Functional/filter/selectivity/plan/q1.res (hunk 3 above)
Comment (rhou1) [inline framework/resources/Functional/filter/selectivity/plan/q1.res]: I removed this test.  It was the base for the other tests.

**Comment by Agirish** [comment]
+1

**Pull request merged**